tests/components/reolink/test_binary_sensor.py

```diff
@@ -0,0 +1,52 @@
+"""Test the Reolink binary sensor platform."""
+
+from datetime import timedelta
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.reolink import DEVICE_UPDATE_INTERVAL, const
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util.dt import utcnow
+
+from .conftest import TEST_NVR_NAME, TEST_UID
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+from tests.typing import ClientSessionGenerator
+
+
+async def test_motion_sensor(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    config_entry: MockConfigEntry,
+    reolink_connect: MagicMock,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test binary sensor entity with motion sensor."""
+    reolink_connect.model = "Reolink Duo PoE"
+    reolink_connect.motion_detected.return_value = True
+    with patch("homeassistant.components.reolink.PLATFORMS", [Platform.BINARY_SENSOR]):
+        assert await hass.config_entries.async_setup(config_entry.entry_id) is True
+    await hass.async_block_till_done()
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    entity_id = f"{Platform.BINARY_SENSOR}.{TEST_NVR_NAME}_motion_lens_0"
+    assert hass.states.is_state(entity_id, "on")
+
+    reolink_connect.motion_detected.return_value = False
+    async_fire_time_changed(
+        hass, utcnow() + DEVICE_UPDATE_INTERVAL + timedelta(seconds=30)
+    )
```

Comment:
```suggestion
    freezer.tick(DEVICE_UPDATE_INTERVAL)
    async_fire_time_changed(hass)
```
Not sure if the 30 seconds have any significance

Reply:
No not specifically, just wanted to make sure the update actually fires.
The diffrence between updating AT 13:00:00.000 or AFTER 13:00:00.000 (so basically 1 ms or us later). Did not know how this was implemented upstream.
And wanted to be rather safe than sorry.

Reply:
afaik its safe to just do AT the moment

Reply:
done